WNPRC_Purchasing/src/client/components/VendorInputModal.tsx

```diff
@@ -0,0 +1,279 @@
+                        />
+                        <NewVendorInput
+                            required={false}
+                            column={'phoneNumber'}
+                            columnTitle={'Phone'}
+                            value={updatedNewVendor.phoneNumber}
+                            onChange={onInputChange}
+                        />
+                        <NewVendorInput
+                            required={false}
+                            column={'faxNumber'}
+                            columnTitle={'Fax'}
+                            value={updatedNewVendor.faxNumber}
+                            onChange={onInputChange}
+                        />
+                        <NewVendorInput
+                            required={false}
+                            column={'email'}
+                            columnTitle={'Email'}
+                            value={updatedNewVendor.email}
+                            onChange={onInputChange}
+                        />
+                        <NewVendorTextArea
+                            required={false}
+                            column={'url'}
+                            columnTitle={'Company website'}
+                            value={updatedNewVendor.url}
+                            onChange={onInputChange}
+                        />
+                        <NewVendorTextArea
+                            required={false}
+                            column={'notes'}
+                            columnTitle={'Notes'}
+                            value={updatedNewVendor.notes}
+                            onChange={onInputChange}
+                        />
+                    </Form>
+                </Modal.Body>
+                <Modal.Footer>
+                    <Button variant="secondary" onClick={handleClose}>
+                        Cancel
+                    </Button>
+                    <Button variant="primary" onClick={onAddingNewVendor}>
+                        Save
+                    </Button>
+                </Modal.Footer>
+                {
+                    (updatedNewVendor.errorMsg) &&
+                    <div className='alert alert-danger'>
+                        {updatedNewVendor.errorMsg}
+                    </div>
+                }
+
+            </Modal>
+        </>
+    );
+}
+
+interface VendorProps
+{
+    required: boolean;
+    column: string;
+    columnTitle: string;
+    value: any;
+    onChange: (column:string, value:any) => void;
+    hasError?: boolean;
+}
+
+const NewVendorInput: FC<VendorProps> = (props) => {
+
+    const {required, column, columnTitle, value, onChange, hasError} = props;
+
+    const onValueChange = useCallback((evt) => {
+        onChange(column, evt.target.value);
+    }, [required, column, columnTitle, value, onChange]);
+
+    return (
+        <div style={{width:'100%'}}>
```

Comment:
We should try to use css classes instead of inline styling

Reply:
done